dask/dataframe/accessor.py

```diff
@@ -119,6 +121,15 @@ def _validate(self, series):
     def split(self, pat=None, n=-1):
         return self._function_map('split', pat=pat, n=n)
 
+    @derived_from(pd.core.strings.StringMethods)
+    def cat(self, others=None, sep=None, na_rep=None):
+        if others is None:
+            raise NotImplementedError("x.str.cat() with `others == None`")
+        elif not isinstance(others, Sequence):
```

Comment:
Pandas seems to expect a non-string sequence here:


```python
In [28]: s = pd.Series(['a', 'b', 'c'])

In [29]: s.str.cat('abc')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/Envs/dask-dev/lib/python3.6/site-packages/pandas/pandas/core/strings.py in _length_check(others)
    146                 n = len(x)
--> 147             elif len(x) != n:
    148                 raise ValueError('All arrays must be same length')

TypeError: len() of unsized object

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-29-e66cbcccce1f> in <module>()
----> 1 s.str.cat('abc')

~/Envs/dask-dev/lib/python3.6/site-packages/pandas/pandas/core/strings.py in cat(self, others, sep, na_rep)
   1473     def cat(self, others=None, sep=None, na_rep=None):
   1474         data = self._orig if self._is_categorical else self._data
-> 1475         result = str_cat(data, others=others, sep=sep, na_rep=na_rep)
   1476         return self._wrap_result(result, use_codes=(not self._is_categorical))
   1477

~/Envs/dask-dev/lib/python3.6/site-packages/pandas/pandas/core/strings.py in str_cat(arr, others, sep, na_rep)
    101         arrays = _get_array_list(arr, others)
    102
--> 103         n = _length_check(arrays)
    104         masks = np.array([isna(x) for x in arrays])
    105         cats = None

~/Envs/dask-dev/lib/python3.6/site-packages/pandas/pandas/core/strings.py in _length_check(others)
    148                 raise ValueError('All arrays must be same length')
    149         except TypeError:
--> 150             raise ValueError("Did you mean to supply a `sep` keyword?")
    151     return n
    152

ValueError: Did you mean to supply a `sep` keyword?

In [30]: dd.from_pandas(s, 2).str.cat('abc')
Out[30]:
Dask Series Structure:
npartitions=1
0    object
2       ...
dtype: object
Dask Name: str_cat, 2 tasks

In [31]: dd.from_pandas(s, 2).str.cat('abc').compute()
Out[31]:
0    aa
1    bb
2    cc
dtype: object
```

We should probably do the same, so that `In[30]` raises.

Reply:
Fixed.